src/renderer/domains/identity/model/identity/model.ts

```diff
@@ -45,55 +44,54 @@ const {
   async fn({ api, accounts }) {
     if (accounts.length === 0) return {};
 
-    const response = await identityPallet.storage.identityOf(api, accounts);
+    const subIdentities = await identityPallet.storage.superOf(api, accounts);
+    const parentAccounts = subIdentities.map(({ account, identity }) => (nullable(identity) ? account : identity[0]));
+    const parentIdentities = await identityPallet.storage.identityOf(api, parentAccounts);
 
-    return response.reduce<IdentityData>((acc, record) => {
-      if (record.identity) {
-        acc[record.account] = {
-          accountId: record.account,
-          name: record.identity[0].info.display,
-          email: record.identity[0].info.email,
-          image: record.identity[0].info.image,
-        };
-      }
+    const result: IdentityData = {};
 
-      return acc;
-    }, {});
+    for (let index = 0; index < parentIdentities.length; index++) {
+      const parent = parentIdentities[index];
+      if (nullable(parent?.identity)) continue;
+
+      const subIdentityName = subIdentities[index]?.identity?.[1] ?? '';
```

Comment:
zipWith?

Reply:
Hm, not clearly understand what you suggest, could you give more verbose explanation please?